### PR TITLE
add redis cache facts

### PR DESCRIPTION
**Pull request by yungezz**
##### SUMMARY

redis cache facts module

##### ISSUE TYPE

- New Module Pull Request

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: lib/ansible/modules/cloud/azure/azure_rm_rediscache_facts.py [hunk 1]
```diff
@@ -0,0 +1,327 @@
...
+        return results
+
+    def serialize_rediscache(self, rediscache):
+        '''
+        Convert a Azure redis cache object to dict.
+        :param cdn: Azure redis cache object
+        :return: dict
+        '''
+        new_result = {}
+        new_result['id'] = rediscache.id
```
Comment (zikalino) [inline lib/ansible/modules/cloud/azure/azure_rm_rediscache_facts.py]: you can create entire dictionary at once

**Comment by yungezz** [comment]
add tests after redis cache merged

**Comment by Fred-sun** [comment]
@yungezz I have tested this PR and get successful result, Does this PR ready for review? could you can remove the WIP label ?

**Comment by Fred-sun** [comment]
@yungezz How about the PR going, Could we can remove the WIP label and push for review? Thanks!

**Comment by Fred-sun** [comment]
@yuwzho @zikalino

File: lib/ansible/modules/cloud/azure/azure_rm_rediscache_facts.py [hunk 2]
```diff
@@ -0,0 +1,340 @@
...
+    resource_group:
+        description:
+            - The resource group to search for the desired Azure redis cache
+        required: True
+    name:
+        description:
+            - Limit results to a specific Azure redis cache.
+    return_access_keys:
+        description:
+            - Indicate wheather to return access keys of the redis cache.
```
Comment (zikalino) [inline lib/ansible/modules/cloud/azure/azure_rm_rediscache_facts.py]: whether

File: lib/ansible/modules/cloud/azure/azure_rm_rediscache_facts.py [hunk 3]
```diff
@@ -0,0 +1,340 @@
...
+
+extends_documentation_fragment:
+    - azure
+
+author:
+    - "Yunge Zhu (@yungezz)"
+'''
+
+EXAMPLES = '''
+    rediscaches:
```
Comment (zikalino) [inline lib/ansible/modules/cloud/azure/azure_rm_rediscache_facts.py]: that should be example of module usage, not return value

File: lib/ansible/modules/cloud/azure/azure_rm_rediscache_facts.py [hunk 4]
```diff
@@ -0,0 +1,340 @@
...
+                - Name of the Azure redis cache.
+            returned: always
+            type: str
+            sample: testRedis
+        id:
+            description:
+                - Id of the Azure redis cache.
+            returned: always
+            type: str
+            sample: /subscriptions/<subs_id>/resourceGroups/<resourcegroup>/providers/Microsoft.Cache/Redis/testredis1
```
Comment (zikalino) [inline lib/ansible/modules/cloud/azure/azure_rm_rediscache_facts.py]: should be consistent with sample name and resource_group

File: lib/ansible/modules/cloud/azure/azure_rm_rediscache_facts.py [hunk 5]
```diff
@@ -0,0 +1,340 @@
...
+                    type: str
+                    sample: C1
+        static_ip:
+            description:
+                - Static IP address.
+            type: str
+            sample: 10.75.0.11
+        subnet:
+            description:
+                - The full resource ID of a subnet in a virtual network to deploy the Redis cache in.
```
Comment (zikalino) [inline lib/ansible/modules/cloud/azure/azure_rm_rediscache_facts.py:129]: should use consistently "redis cache" or "Redis cache" upper case / lower case?

**Review by zikalino — CHANGES_REQUESTED** [review]
the main comment is about examples, others are minor.

**Comment by yungezz** [comment]
ready_for_review

**Review by zikalino — APPROVED** [review]

**Pull request merged**